kernel/init.c

```diff
@@ -302,11 +340,19 @@ static void prepare_multithreading(struct k_thread *dummy_thread)
 	_add_thread_to_ready_q(_main_thread);
 
 #ifdef CONFIG_MULTITHREADING
-	_setup_new_thread(_idle_thread, _idle_stack,
-			  IDLE_STACK_SIZE, idle, NULL, NULL, NULL,
-			  K_LOWEST_THREAD_PRIO, K_ESSENTIAL);
-	_mark_thread_as_started(_idle_thread);
-	_add_thread_to_ready_q(_idle_thread);
+	init_idle_thread(_idle_thread, _idle_stack);
+#endif
+
+#if defined(CONFIG_SMP) && CONFIG_MP_NUM_CPUS > 1
+	init_idle_thread(_idle_thread1, _idle_stack1);
+#endif
+
+#if defined(CONFIG_SMP) && CONFIG_MP_NUM_CPUS > 2
+	init_idle_thread(_idle_thread2, _idle_stack2);
+#endif
+
+#if defined(CONFIG_SMP) && CONFIG_MP_NUM_CPUS > 3
+	init_idle_thread(_idle_thread3, _idle_stack3);
 #endif
```

Comment:
all the #if should be under the CONFIG_MULTITHREADING

Reply:
CONFIG_SMP presumes CONFIG_MULTITHREADING.  It doesn't make sense otherwise.  Not sure if the kconfig makes that explicit or not.